### PR TITLE
Lesson1.8

### PR DESCRIPTION
**Pull request by Miminiverse**
Hello Kasozi, 
I have tried to implement the DELETE method as well from Airtable, and I successfully managed to remove the items. However, the UI doesn't update simultaneously as I press the remove button. Only when I reload the page that the item will actually be removed. 
Can you please check it for me and let me know how I can fix this issue.
Thank you!

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -2,59 +2,93 @@
...
+          id: todo.id
+        }
+
+      })
+      setTodoList(todos)
+      setIsLoading(false)
+
+    } catch (error) {
+      console.log(error)
     }
```
Comment (kwilson05) [inline src/App.js:40]: 💯 Wow!! Super impressive here. Looks like professional code here with the `map` function :)

File: src/App.js [hunk 2]
```diff
@@ -2,59 +2,93 @@
...
   const addTodo = (newTodo) => {
-    setTodoList((prevList) => [...prevList, newTodo]);
-  };
+    fetch (url, {
+      method: 'POST',
+      headers: {  
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_KEY}`,
+    },
+      body: JSON.stringify({fields: newTodo})
```
Comment (kwilson05) [inline src/App.js:54]: You'll need to change the `title` prop here to `Title` to get this working. 👍

File: src/App.js [hunk 3]
```diff
@@ -2,59 +2,93 @@
...
+    console.log(todo)
+    const urlDel = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}/` + todo.id
+    const options ={
+      method: 'DELETE',
+      headers: {
+        Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_KEY}`,
+    }
+  }  
+    const response = await fetch (urlDel, options)
+    setTodoList((prevList) => prevList.filter((item) => item.id !== todo.id))
```
Comment (kwilson05) [inline src/App.js:79]: 💯 Looks like you figured it out 🎖️

File: src/App.js [hunk 4]
```diff
@@ -2,59 +2,93 @@
...
+      setTodoList((prevList) => [...prevList, addTodo])
+    })
+    }
 
 
-  function removeTodo (todo) {
-    setTodoList((newArray) => newArray.filter((list) => todo.id !== list.id))
+  const removeTodo = async (todo) => {
+    console.log(todo)
+    const urlDel = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}/` + todo.id
```
Comment (kwilson05) [inline src/App.js:71]: You could actually include the `todo.id` in the template string like this:
```suggestion
    const urlDel = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}/${todo.id}`
```

File: src/App.js (hunk 3 above)
Comment (kwilson05) [reply src/App.js:79]: One thing I might add here while you're at it is checking if the `DELETE` actually was successful. 
See if you steal how you checked if the request successful from initial load. Then filter on successful `DELETE` or show some type of error message in the console 👍

**Review by kwilson05 — APPROVED** [review]
💯 Good job going above and beyond on this exercise

File: src/App.js (hunk 2 above)
Comment (Miminiverse) [reply src/App.js:54]: Hi Kasozi, I don't really get why I need to change the title prop here to Title? It's working fine with title, can you explain a bit more?

Comment (kwilson05) [reply src/App.js:54]: Hmm maybe your Airtable column name is set up differently than mine but my table column's name is `Title` not `title`. So I had to change it to add a new todo item :)

**Pull request merged**